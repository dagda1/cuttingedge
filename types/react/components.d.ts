import type { ComponentType } from 'react';
export declare function getDisplayName<T>(WrappedComponent: ComponentType<T>): string;
//# sourceMappingURL=components.d.ts.map