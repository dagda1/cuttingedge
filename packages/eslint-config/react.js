module.exports = {
  extends: './index',
  parserOptions: {
    ecmaVersion: 2018,
    sourceType: 'module',
    ecmaFeatures: {
      jsx: true,
    },
  },
  extends: ['./index', 'plugin:react/recommended', 'plugin:jsx-a11y/recommended'],
  settings: {
    react: {
      pragma: 'React',
      version: 'detect',
    },
  },
  parserOptions: {
    ecmaFeatures: {
      jsx: true,
    },
  },
  plugins: ['react', 'react-hooks', 'jest', 'jest-formatting'],
  rules: {
    'react-hooks/rules-of-hooks': [
      'error',
      {
        additionalHooks: '(useIsomorphicLayoutEffect)',
      },
    ],
    'react-hooks/exhaustive-deps': [
      'error',
      {
        additionalHooks: '(useIsomorphicLayoutEffect)',
      },
    ],
    'react/prop-types': 0,
    'react/react-in-jsx-scope': 0,
  },
};
