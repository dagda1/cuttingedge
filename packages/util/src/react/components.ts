import React from 'react';

export function getDisplayName<T>(WrappedComponent: React.ComponentType<T>): string {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component';
}
