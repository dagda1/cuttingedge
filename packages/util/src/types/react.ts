import type { DetailedHTMLProps, HTMLAttributes, FC, ComponentClass } from 'react';

export type Taggable<TProps = DetailedHTMLProps<HTMLAttributes<HTMLElement>, HTMLElement>> =
  | FC<TProps>
  | ComponentClass<TProps>
  | string;
