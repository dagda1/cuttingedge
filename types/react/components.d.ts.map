{"version":3,"file":"components.d.ts","sourceRoot":"","sources":["../../packages/util/src/react/components.ts"],"names":[],"mappings":"AAAA,OAAO,KAAK,EAAE,aAAa,EAAE,MAAM,OAAO,CAAC;AAE3C,wBAAgB,cAAc,CAAC,CAAC,EAAE,gBAAgB,EAAE,aAAa,CAAC,CAAC,CAAC,GAAG,MAAM,CAE5E"}