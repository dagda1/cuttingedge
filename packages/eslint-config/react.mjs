import react from "eslint-plugin-react";
import reactHooks from "eslint-plugin-react-hooks";
import { fixupPluginRules } from "@eslint/compat";
import path from "node:path";
import { fileURLToPath } from "node:url";
import js from "@eslint/js";
import { FlatCompat } from "@eslint/eslintrc";
import defaultConfig from './index.mjs'

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const compat = new FlatCompat({
    baseDirectory: __dirname,
    recommendedConfig: js.configs.recommended,
    allConfig: js.configs.all
});

export default [
    ...compat.extends("plugin:react/recommended", "plugin:jsx-a11y/recommended"),
    ...defaultConfig,
    {
        plugins: {
            react,
            "react-hooks": fixupPluginRules(reactHooks),
        },

        languageOptions: {
            ecmaVersion: 5,
            sourceType: "script",

            parserOptions: {
                ecmaFeatures: {
                    jsx: true,
                },
            },
        },

        settings: {
            react: {
                pragma: "React",
                version: "detect",
            },
        },

        rules: {
            "react-hooks/rules-of-hooks": ["error"],

            "react-hooks/exhaustive-deps": ["error", {
                additionalHooks: "(useIsomorphicLayoutEffect)",
            }],

            "react/prop-types": 0,
            "react/react-in-jsx-scope": 0,
        },
    },
];