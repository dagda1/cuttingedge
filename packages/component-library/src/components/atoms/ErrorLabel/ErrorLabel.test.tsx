import { expect, it, describe } from '@jest/globals';
import { ErrorLabel, ErrorProps } from '.';
import { render } from '@testing-library/react';

const wrap = (props: ErrorProps) => render(<ErrorLabel {...props} />);

describe('ErrorLabel', () => {
  it('should tag custom data-selector', () => {
    const { getByTestId } = wrap({
      id: 'error',
      dataSelector: 'error-selector',
      errorMessage: 'Error',
    });

    const label = getByTestId('error-selector') as HTMLLabelElement;

    expect(label.innerHTML).toBe('Error');
  });
});
