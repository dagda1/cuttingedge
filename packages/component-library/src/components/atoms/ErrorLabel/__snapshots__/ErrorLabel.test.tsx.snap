// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorLabel renders consistently 1`] = `
<div
  aria-hidden="false"
  className="form-group"
  id="error"
>
  <ul
    className="current-errors container"
  >
    <li
      data-selector="form-error"
    >
      error
    </li>
  </ul>
</div>
`;
